lms/djangoapps/program_enrollments/api/writing.py

```diff
@@ -424,3 +449,52 @@ def _organize_requests_by_external_key(enrollment_requests):
             continue
         requests_by_key[key] = request
     return requests_by_key, duplicated_keys
+
+
+def _process_conflicted_course_enrollments(
+    requests_by_key,
+    existing_course_enrollments,
+    program_uuid,
+    course_key
+):
+    """
+    Process the list of existing course enrollments together with
+    the enrollment request list stored in 'requests_by_key'. Detect
+    whether we have conflicted ACTIVE ProgramCourseEnrollment entries.
+    When detected, log about it and set that update or create entry to
+    have duplicated status.
+
+    Arguments:
+        requests_by_key (dict)
+        existing_course_enrollments (queryset),
+        program_uuid (UUID|str),
+        course_key (str)
+
+    Returns:
+        results (dict) with detected conflict entry, or empty dict.
+    """
+    results = {}
+
+    course_enrollment_statuses_by_user_key = {
+        key: request.get('status') for key, request in requests_by_key.items()
+    }
+
+    for course_enrollment in existing_course_enrollments:
+        external_user_key = course_enrollment.program_enrollment.external_user_key
+        enrollment_status = course_enrollment_statuses_by_user_key.get(
+            course_enrollment.program_enrollment.external_user_key
+        )
+        if enrollment_status and \
+            enrollment_status == ProgramCourseEnrollmentStatuses.ACTIVE and \
+            course_enrollment.status == ProgramCourseEnrollmentStatuses.ACTIVE and \
+                str(course_enrollment.program_enrollment.program_uuid) != str(program_uuid):
```

Comment:
Nit: how about a little helper function like `_is_conflicted_enrollment(program_uuid, course_enrollment, enrollment_status)`?

Reply:
After the suggestion from Kyle of having is_active property, this is much easier to read.